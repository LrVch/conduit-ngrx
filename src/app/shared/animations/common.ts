import {
  animate,
  query,
  style,
  transition,
  trigger,
  stagger,
  sequence,
  animation,
  group,
  useAnimation,
  state,
  animateChild
} from '@angular/animations';
import {
  AnimationsService,
  RouteAnimationChangeType
} from '@app/core/services/animations.service';

export const scaleOut = animation(
  [
    style({ opacity: 0, width: 0, 'min-width': 0, transform: 'scale(0,1)' }),
    group([
      animate('{{timeShort}}', style({ width: '*' })),
      animate('{{timeLong}}', style({ opacity: 1, transform: 'scale(1,1)' }))
    ])
  ],
  {
    params: {
      timeShort: '100ms ease-out',
      timeLong: '200ms ease-out'
    }
  }
);

export const scaleIn = animation(
  [
    style({ 'min-width': 0 }),
    group([
      animate('{{timeShort}}', style({ opacity: 0, transform: 'scale(0,1)' })),
      animate('{{timeLong}}', style({ width: 0 }))
    ])
  ],
  {
    params: {
      timeShort: '100ms ease-out',
      timeLong: '200ms ease-out'
    }
  }
);

export const expand = animation(
  [
    sequence([
      style({ opacity: 0, height: 0 }),
      animate('{{duration}} {{timing}}', style({ height: '*' })),
      animate('{{duration}} {{timing}}', style({ opacity: 1 }))
    ])
  ],
  {
    params: {
      duration: '300ms',
      timing: 'ease-out'
    }
  }
);

export const collapce = animation(
  [
    sequence([
      animate('{{duration}}', style({ opacity: 0 })),
      animate('{{duration}} {{timing}}', style({ height: 0 }))
    ])
  ],
  {
    params: {
      duration: '200ms',
      timing: 'ease-out'
    }
  }
);

export const avatarInOut = trigger('avatarInOut', [
  transition(':enter', [useAnimation(scaleOut)]),
  transition(':leave', [useAnimation(scaleIn)])
]);

export const commentAnimation = trigger('commentAnimation', [
  transition(':enter', [useAnimation(expand)]),
  transition(':leave', [useAnimation(collapce)])
]);

export const swingChange = animation(
  [
    sequence([
      query(':enter', [style({ opacity: 0, position: 'absolute' })]),
      query(':leave', [
        style({ transform: '{{transform}}(0)' }),
        animate(
          '{{duration}} {{timing}}',
          style({ opacity: 0, transform: '{{transform}}(100%)' })
        )
      ]),
      query(':leave', [style({ position: 'absolute' })]),
      query(':enter', [
        style({
          opacity: 0,
          transform: '{{transform}}({{direction}}100%)',
          position: 'static'
        }),
        animate(
          '{{duration}} {{timing}}',
          style({ opacity: 1, transform: '{{transform}}(0)' })
        )
      ])
    ])
  ],
  {
    params: {
      duration: '150ms',
      timing: 'cubic-bezier(0.35, 0, 0.25, 1)',
      transform: 'translateY',
      direction: '-'
    }
  }
);

export const deleteButtonAnimation = trigger('deleteButtonAnimation', [
  transition('false <=> true', [useAnimation(swingChange)])
]);

export const slidePageAnimation = animation(
  [
    style({ position: 'relative', overflow: 'hidden' }),
    query(':enter > *', [style({ opacity: 0, position: 'absolute' })], {
      optional: true
    }),
    query(':leave', animateChild(), {
      optional: true
    }),
    sequence([
      query(
        ':leave > *',
        [
          style({
            transform: 'translateX(0)',
            opacity: 1,
            position: 'relative'
          }),
          animate(
            '{{duration}} {{timing}}',
            style({ transform: 'translateX({{distanse}})', opacity: 0 })
          )
        ],
        {
          optional: true
        }
      ),
      query(':leave > *', style({ position: 'absolute' }), {
        optional: true
      }),
      query(
        ':enter > *',
        [
          style({
            position: 'relative',
            transform: 'translateX(-{{distanse}})'
          }),
          animate(
            '{{duration}} {{delay}} {{timing}}',
            style({ transform: 'translateX(0)', opacity: 1 })
          )
        ],
        {
          optional: true
        }
      )
    ]),
    query(':enter', animateChild(), {
      optional: true
    })
  ],
  {
    params: {
      distanse: '30px',
      duration: '300ms',
      delay: '300ms',
      timing: 'ease-out'
    }
  }
);

export const fadePageAnimation = animation(
  [
    style({ position: 'relative', overflow: 'hidden' }),
    query(':enter > *', [style({ opacity: 0, position: 'absolute' })], {
      optional: true
    }),
    query(':leave', animateChild(), {
      optional: true
    }),
    sequence([
      query(
        ':leave > *',
        [
          style({ opacity: 1, position: 'relative' }),
          animate('{{duration}} {{timing}}', style({ opacity: 0 }))
        ],
        {
          optional: true
        }
      ),
      query(':leave > *', style({ position: 'absolute' }), {
        optional: true
      }),
      query(
        ':enter > *',
        [
          style({ position: 'relative' }),
          animate('{{duration}} {{timing}}', style({ opacity: 1 }))
        ],
        {
          optional: true
        }
      )
    ]),
    query(':enter', animateChild(), {
      optional: true
    })
  ],
  {
    params: {
      distanse: '30px',
      duration: '300ms',
      timing: 'ease-out'
    }
  }
);

export function staticPages(fromState: string, toState: string) {
  return (
    toState === 'SettingsComponent' ||
    toState === 'EditorComponent' ||
    toState === 'login' ||
    toState === 'register'
  );
}

export function slideAnimation(fromState, toState) {
  if (!AnimationsService.isRouteAnimationsEnabled()) {
    return false;
  }

  if (!AnimationsService.isRouteAnimationsChangeType('SLIDE')) {
    return false;
  }

  return staticPages(fromState, toState);
}

export function fadeAnimation(fromState, toState) {
  if (!AnimationsService.isRouteAnimationsEnabled()) {
    return false;
  }

  if (!AnimationsService.isRouteAnimationsChangeType('FADE')) {
    return false;
  }

  return staticPages(fromState, toState);
}

export function typeAnimation(type: RouteAnimationChangeType, pages) {
  return state;
}

export const routeAnimation = trigger('routeAnimation', [
  transition(slideAnimation, [useAnimation(slidePageAnimation)]),
  transition(fadeAnimation, [useAnimation(fadePageAnimation)])
]);
