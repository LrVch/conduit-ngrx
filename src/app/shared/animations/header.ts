import {
    animate,
    query,
    style,
    transition,
    trigger,
    stagger,
    sequence
} from '@angular/animations';

export const a = '';
