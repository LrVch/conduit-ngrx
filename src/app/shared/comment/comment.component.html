<mat-card class="comment">
  <mat-progress-bar class="comment__loader"
                    [hidden]="!comment.isDeleting"
                    mode="indeterminate">
  </mat-progress-bar>
  <mat-card-content>
    <p>
      {{comment.body}}
    </p>
    <mat-icon [ngStyle]="{'opacity': comment.isDeleting ? 0.5 : 1}"
              [hidden]="!canModify"
              class="comment__delete"
              (click)="deleteComment($event)"
              aria-label="delete">delete</mat-icon>
  </mat-card-content>
  <mat-card-header>
    <div mat-card-avatar
         class="comment__avatar">
      <a [routerLink]="['/profile', comment.author.username]">
        <img [src]="comment.author.image"
             [attr.alt]="comment.author.username">
      </a>
    </div>
    <mat-card-title>
      <a class="comment__author"
         [routerLink]="['/profile', comment.author.username]">{{ comment.author.username }}</a>
    </mat-card-title>
    <mat-card-subtitle>
      {{ comment.createdAt | date: 'medium' }}
    </mat-card-subtitle>
  </mat-card-header>
</mat-card>