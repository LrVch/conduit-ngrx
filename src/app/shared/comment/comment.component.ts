import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Comment, User } from 'src/app/core';

@Component({
  selector: 'app-comment',
  templateUrl: './comment.component.html',
  styleUrls: ['./comment.component.scss']
})
export class CommentComponent {
  @Input('comment') comment: Comment;
  @Input('user') user: User;
  @Input('canModify') canModify: boolean;
  @Output() delete = new EventEmitter<boolean>();

  deleteComment(): void {
    if (this.comment.isDeleting) {
      return;
    }

    this.delete.emit(true);
  }
}
