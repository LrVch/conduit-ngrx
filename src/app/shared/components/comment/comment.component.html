<mat-card class="comment">
  <mat-progress-bar class="comment__loader"
                    [hidden]="!comment.isDeleting"
                    mode="indeterminate">
  </mat-progress-bar>
  <mat-card-content>
    <p>
      {{comment.body}}
    </p>
    <mat-icon [ngStyle]="{'opacity': comment.isDeleting ? 0.5 : 1}"
              [hidden]="!canModify"
              class="comment__delete"
              (click)="deleteComment()"
              aria-label="delete">delete</mat-icon>
  </mat-card-content>
  <mat-card-header>
    <div mat-card-avatar
         class="comment__avatar">
      <a [routerLink]="['/profile', comment.author.username]">
        <div>
          <img [src]="comment.author.image"
               appAvatar
               [attr.alt]="comment.author.username">
        </div>
      </a>
    </div>
    <mat-card-title>
      <a class="comment__author"
         [routerLink]="['/profile', comment.author.username]">{{ comment.author.username }}</a>
    </mat-card-title>
    <mat-card-subtitle>
      {{ comment.createdAt | date: 'short' : '' : locale }}
    </mat-card-subtitle>
  </mat-card-header>
</mat-card>