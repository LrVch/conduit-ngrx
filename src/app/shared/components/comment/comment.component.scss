:host {
    display: block;
    margin-bottom: 30px;
    &:last-child {
        margin-bottom: 0;
    }
}

.comment {
    .mat-card-content {
        padding-right: 60px;
    }
}

.comment__avatar {
    display: flex;
    overflow: hidden;
    a {
        display: flex;
    }
    img {
        max-width: 100%;
    }
}

a.comment__author {
    color: #000;
    text-decoration: none;
    &:hover {
        opacity: .8;
    }
}

.comment__delete {
    cursor: pointer;
    position: absolute;
    right: 10px;
    top: 10px;
}

.comment__loader {
    position: absolute;
    top: 0;
    right: 0;
}