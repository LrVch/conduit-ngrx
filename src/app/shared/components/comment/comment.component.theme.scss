@import '~@angular/material/theming';

@mixin conduit-comment-component-theme($theme) {
  $primary: map-get($theme, primary);
  $accent: map-get($theme, accent);
  $foreground: map-get($theme, foreground);
  $background: map-get($theme, background);

  app-comment {
    mat-card-title {
        .comment__author {
            color: mat-color($foreground, text);
        }
    }
  }
}