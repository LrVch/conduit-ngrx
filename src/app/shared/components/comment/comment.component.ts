import {
  Component,
  Input,
  Output,
  EventEmitter,
  HostBinding
} from '@angular/core';
import { Comment } from '@app/core';
import { commentAnimation } from '@app/shared/animations';

@Component({
  selector: 'app-comment',
  templateUrl: './comment.component.html',
  styleUrls: ['./comment.component.scss'],
  animations: [commentAnimation]
})
export class CommentComponent {
  @Input('locale') locale = 'en';
  @Input('comment') comment: Comment = {} as Comment;
  @Input('canModify') canModify = false;
  @Output() delete = new EventEmitter<boolean>();

  @HostBinding('@commentAnimation') true;

  deleteComment(): void {
    if (this.comment.isDeleting) {
      return;
    }

    this.delete.emit(true);
  }
}
