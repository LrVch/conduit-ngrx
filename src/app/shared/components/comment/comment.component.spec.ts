import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests, getComment } from '@app/lib/testing';

import { CommentComponent } from './comment.component';
import {
  DebugElement,
  Component,
  Input,
  Output,
  EventEmitter
} from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';
import { RouterLinkDirectiveStubDirective } from '@app/lib/testing/directive/router-link-directive-stub';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

@Component({
  selector: 'app-delete-btn',
  template: '<div id="delete" (click)="onDelete()"></div>'
})
class HostDeleteBtnComponent {
  @Input('isDeleting') isDeleting: boolean;
  @Input('showConfirm') showConfirm: boolean;
  @Output() delete = new EventEmitter();

  onDelete() {
    this.delete.emit();
  }
}

describe('CommentComponent', () => {
  let component: CommentComponent;
  let fixture: ComponentFixture<CommentComponent>;
  let de: DebugElement;
  let el: HTMLElement;
  let linkDes;
  let routerLinks;
  let comment;

  beforeEach(async(() => {
    comment = getComment();
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [
          CommentComponent,
          RouterLinkDirectiveStubDirective,
          HostDeleteBtnComponent
        ],
        imports: [MaterialModule, NoopAnimationsModule]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(CommentComponent);
      component = fixture.componentInstance;
      component.comment = comment;
      de = fixture.debugElement;
      el = de.nativeElement;
      fixture.detectChanges();
      linkDes = de.queryAll(By.directive(RouterLinkDirectiveStubDirective));
      routerLinks = linkDes.map(deel =>
        deel.injector.get(RouterLinkDirectiveStubDirective)
      );
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should has "comment" @Input', () => {
    expect(component.comment).toEqual(comment);
  });

  it('should has "canModify" @Input', () => {
    expect(component.canModify).toEqual(false);
  });

  it('should show "comment" @Input', () => {
    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should show "isDeleting" loader via comment.isDeleting', () => {
    comment.isDeleting = true;

    fixture.detectChanges();
    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should show "canModify" @Input', () => {
    component.canModify = true;

    fixture.detectChanges();
    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should raises delete event when clicked (deleteComment)', () => {
    let raised = null;
    component.delete.subscribe(_ => (raised = true));

    fixture.detectChanges();

    const button = de.query(By.css('#delete'));
    button.triggerEventHandler('click', null);

    expect(raised).toBe(true);
  });

  it('shouldn\'t raises delete event when clicked (deleteComment) and "comment.isDeleting" equals to true', () => {
    let wasEmited = false;

    comment.isDeleting = true;
    component.delete.subscribe(_ => (wasEmited = true));

    fixture.detectChanges();
    const button = de.query(By.css('.comment__delete'));

    button.triggerEventHandler('click', null);

    expect(wasEmited).toBe(false);
  });

  it('can get RouterLinks from template', () => {
    expect(routerLinks.length).toBe(2);
    expect(routerLinks[0].linkParams).toEqual([
      '/profile',
      comment.author.username
    ]);
    expect(routerLinks[1].linkParams).toEqual([
      '/profile',
      comment.author.username
    ]);
  });

  it('can click Comment link in template', () => {
    const goToProfileDe = linkDes[0];
    const goToProfileLink = routerLinks[0];

    expect(goToProfileLink.navigatedTo).toBeNull();

    fixture.detectChanges();
    goToProfileDe.triggerEventHandler('click', null);

    expect(goToProfileLink.navigatedTo).toEqual([
      '/profile',
      comment.author.username
    ]);
  });
});
