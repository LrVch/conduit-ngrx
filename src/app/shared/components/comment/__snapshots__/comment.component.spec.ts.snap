// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentComponent should show "canModify" @Input 1`] = `
<app-comment
  canModify={[Function Boolean]}
  comment={[Function Object]}
  delete={[Function EventEmitter]}
>
  <mat-card
    class="comment mat-card"
  >
    <mat-progress-bar
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="comment__loader mat-progress-bar mat-primary"
      hidden=""
      mode="indeterminate"
      ng-reflect-mode="indeterminate"
      role="progressbar"
    >
      <svg
        class="mat-progress-bar-background mat-progress-bar-element"
        focusable="false"
        height="4"
        width="100%"
      >
        <defs>
          <pattern
            height="4"
            id="mat-progress-bar-5"
            patternUnits="userSpaceOnUse"
            width="8"
            x="4"
            y="0"
          >
            <circle
              cx="2"
              cy="2"
              r="2"
            />
          </pattern>
        </defs>
        <rect
          fill="url('/#mat-progress-bar-5')"
          height="100%"
          width="100%"
        />
      </svg>
      <div
        class="mat-progress-bar-buffer mat-progress-bar-element"
      />
      <div
        class="mat-progress-bar-primary mat-progress-bar-fill mat-progress-bar-element"
        ng-reflect-ng-style="[object Object]"
        style="transform: scaleX(0);"
      />
      <div
        class="mat-progress-bar-secondary mat-progress-bar-fill mat-progress-bar-element"
      />
    </mat-progress-bar>
    <mat-card-content
      class="mat-card-content"
    >
      <p>
         body 
      </p>
      <mat-icon
        aria-hidden="true"
        aria-label="delete"
        class="comment__delete mat-icon material-icons"
        ng-reflect-ng-style="[object Object]"
        role="img"
        style="opacity: 1;"
      >
        delete
      </mat-icon>
    </mat-card-content>
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="comment__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <div>
            <img
              alt="username"
              appavatar=""
              src="image"
            />
          </div>
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="comment__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
           Dec 14, 2018, 4:45:39 PM 
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card>
</app-comment>
`;

exports[`CommentComponent should show "comment" @Input 1`] = `
<app-comment
  canModify="false"
  comment={[Function Object]}
  delete={[Function EventEmitter]}
>
  <mat-card
    class="comment mat-card"
  >
    <mat-progress-bar
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="comment__loader mat-progress-bar mat-primary"
      hidden=""
      mode="indeterminate"
      ng-reflect-mode="indeterminate"
      role="progressbar"
    >
      <svg
        class="mat-progress-bar-background mat-progress-bar-element"
        focusable="false"
        height="4"
        width="100%"
      >
        <defs>
          <pattern
            height="4"
            id="mat-progress-bar-3"
            patternUnits="userSpaceOnUse"
            width="8"
            x="4"
            y="0"
          >
            <circle
              cx="2"
              cy="2"
              r="2"
            />
          </pattern>
        </defs>
        <rect
          fill="url('/#mat-progress-bar-3')"
          height="100%"
          width="100%"
        />
      </svg>
      <div
        class="mat-progress-bar-buffer mat-progress-bar-element"
      />
      <div
        class="mat-progress-bar-primary mat-progress-bar-fill mat-progress-bar-element"
        ng-reflect-ng-style="[object Object]"
        style="transform: scaleX(0);"
      />
      <div
        class="mat-progress-bar-secondary mat-progress-bar-fill mat-progress-bar-element"
      />
    </mat-progress-bar>
    <mat-card-content
      class="mat-card-content"
    >
      <p>
         body 
      </p>
      <mat-icon
        aria-hidden="true"
        aria-label="delete"
        class="comment__delete mat-icon material-icons"
        hidden=""
        ng-reflect-ng-style="[object Object]"
        role="img"
        style="opacity: 1;"
      >
        delete
      </mat-icon>
    </mat-card-content>
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="comment__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <div>
            <img
              alt="username"
              appavatar=""
              src="image"
            />
          </div>
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="comment__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
           Dec 14, 2018, 4:45:39 PM 
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card>
</app-comment>
`;

exports[`CommentComponent should show "isDeleting" loader via comment.isDeleting 1`] = `
<app-comment
  canModify="false"
  comment={[Function Object]}
  delete={[Function EventEmitter]}
>
  <mat-card
    class="comment mat-card"
  >
    <mat-progress-bar
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="comment__loader mat-progress-bar mat-primary"
      mode="indeterminate"
      ng-reflect-mode="indeterminate"
      role="progressbar"
    >
      <svg
        class="mat-progress-bar-background mat-progress-bar-element"
        focusable="false"
        height="4"
        width="100%"
      >
        <defs>
          <pattern
            height="4"
            id="mat-progress-bar-4"
            patternUnits="userSpaceOnUse"
            width="8"
            x="4"
            y="0"
          >
            <circle
              cx="2"
              cy="2"
              r="2"
            />
          </pattern>
        </defs>
        <rect
          fill="url('/#mat-progress-bar-4')"
          height="100%"
          width="100%"
        />
      </svg>
      <div
        class="mat-progress-bar-buffer mat-progress-bar-element"
      />
      <div
        class="mat-progress-bar-primary mat-progress-bar-fill mat-progress-bar-element"
        ng-reflect-ng-style="[object Object]"
        style="transform: scaleX(0);"
      />
      <div
        class="mat-progress-bar-secondary mat-progress-bar-fill mat-progress-bar-element"
      />
    </mat-progress-bar>
    <mat-card-content
      class="mat-card-content"
    >
      <p>
         body 
      </p>
      <mat-icon
        aria-hidden="true"
        aria-label="delete"
        class="comment__delete mat-icon material-icons"
        hidden=""
        ng-reflect-ng-style="[object Object]"
        role="img"
        style="opacity: 0.5;"
      >
        delete
      </mat-icon>
    </mat-card-content>
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="comment__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <div>
            <img
              alt="username"
              appavatar=""
              src="image"
            />
          </div>
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="comment__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
           Dec 14, 2018, 4:45:39 PM 
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card>
</app-comment>
`;
